backend/siarnaq/api/compete/views.py

```diff
@@ -450,6 +458,91 @@ def historical_rating(self, request, pk=None, *, episode_id):
         results = HistoricalRatingSerializer(grouped.values(), many=True).data
         return Response(results, status=status.HTTP_200_OK)
 
+    @extend_schema(
+        parameters=[
+            OpenApiParameter(
+                name="team_id",
+                type=int,
+                description="A team to filter for. Defaults to your own team.",
+            ),
+            OpenApiParameter(
+                name="scrimmage_type",
+                enum=["ranked", "unranked", "all"],
+                default="all",
+                description="Which type of scrimmages to filter for. Defaults to all.",
+            ),
+        ],
+        responses={
+            status.HTTP_200_OK: ScrimmageRecordSerializer(),
+            status.HTTP_400_BAD_REQUEST: OpenApiResponse(
+                description="No team found with the given ID."
+            ),
+        },
+    )
+    @action(
+        detail=False,
+        methods=["get"],
+        permission_classes=(IsEpisodeMutable,),
+    )
+    def scrimmaging_record(self, request, pk=None, *, episode_id):
+        """List the scrimmaging win-loss-tie record of a team."""
+        queryset = self.get_queryset().filter(tournament_round__isnull=True)
+
+        scrimmage_type = self.request.query_params.get("scrimmage_type")
+        if scrimmage_type is not None:
+            if scrimmage_type == "ranked":
+                queryset = queryset.filter(is_ranked=True)
+            elif scrimmage_type == "unranked":
+                queryset = queryset.filter(is_ranked=False)
+
+        team_id = parse_int(self.request.query_params.get("team_id"))
+        if team_id is None and request.user.pk is not None:
+            team_id = (
+                Team.objects.filter(members__pk=request.user.pk)
+                .filter(episode_id=episode_id)
+                .first()
+                .id
+            )
+            if team_id is None:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+        if team_id is not None:
+            queryset = queryset.filter(participants__team=team_id)
+        else:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        has_invisible = self.get_queryset().filter(
+            participants__team__status=TeamStatus.INVISIBLE
+        )
+        queryset = queryset.exclude(pk__in=Subquery(has_invisible.values("pk")))
+
```

Comment:
see if we can improve this whole function's performance using the code here 
https://github.com/battlecode/galaxy/pull/558

Reply:
We can do this when we address #826 :sunglasses: